### DISCUSSION TITLE
Quick and dirty way to do a "HV unbrick" once UPDI pin has been disabled

### DISCUSSION THREAD

**Discussion by hmeijdam** [SpenceKonde/megaTinyCore#450] [Show and tell]
I wanted to know if it's worth the effort to build a HV programmer besides the serial programmer that I use as UPDI programmer. So I softbricked a Attiny412 by turning it's UPDI pin into a GPIO pin.

![UPDI disabled](https://user-images.githubusercontent.com/9595465/121197036-420dea80-c871-11eb-8fe3-9b3745fe99f3.jpg)

After a power cycle I confirmed that it was no longer possible to upload anything to it via UPDI. Then I took a 12 volt alkaline battery, disconnected the UPDI program wire (it doesn't like 12v I guess) powercycled the Attiny and then briefly touched it's UPDI pin with the 12V output from the little battery.

![HV programmer](https://user-images.githubusercontent.com/9595465/121197634-c1032300-c871-11eb-86e5-ddb393d3e0db.jpg)
After that I can connect the UPDI program wire again and use it for loading bootloader and/or sketches. 

Touching the UPDI pin with 12V turned out not too time critical, but leaving it on for about more than half a second did not work. Apparently there is a certain window where the HV pulse works. In the datasheet it is mentioned as between 100 us and one ms, but I think I left it on for waaaaay longer than one ms. Or maybe it was the bouncing if the mechanical connection with the pin that did the real work.
It actually worked so well (on this AT412 part at least) that I'm not going to build a HV programmer for now. 

Just my 2 cents

[edit] If I really wanted to know why/how this works I would have to make that HV programmer and play with the pulselength of the HV pulse.

**Comment by SpenceKonde** [comment]
Yeah, I've heard of this trick. there was some problem with it - my vague recollection was that it worked on 8/14pin parts, but then 20/24 pin parts need a KEY command sent almost immediately or something. We were paying attention to it ab out a year or two ago IIRC
